test/lib/jdk/test/lib/cds/CDSArchiveUtils.java

```diff
@@ -0,0 +1,308 @@
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package jdk.test.lib.cds;
+
+import java.io.IOException;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Random;
+
+import java.nio.file.Files;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.nio.channels.FileChannel;
+import static java.nio.file.StandardCopyOption.REPLACE_EXISTING;
+import java.nio.file.StandardOpenOption;
+import static java.nio.file.StandardOpenOption.READ;
+import static java.nio.file.StandardOpenOption.WRITE;
+
+import jdk.test.lib.Utils;
+import sun.hotspot.WhiteBox;
+
+// This class performs operations on shared archive file
+public class CDSArchiveUtils {
+    public static int offsetMagic;           // CDSFileMapHeaderBase::_magic
+    public static int offsetVersion;         // CDSFileMapHeaderBase::_version
+    public static int offsetJvmIdent;        // FileMapHeader::_jvm_ident
+    public static int spOffsetCrc;           // CDSFileMapRegion::_crc
+    public static int fileHeaderSize;        // total size of header, aligned with alignment
+    public static int cdsFileMapRegionSize;  // size of CDSFileMapRegion
+    public static int spOffset;              // offset of CDSFileMapRegion
+    public static int spUsedOffset;          // offset of CDSFileMapRegion::_used
+    public static int sizetSize;             // size of size_t
+    public static int intSize;               // size of int
+    public static long alignment;            // MetaspaceShared::core_region_alignment
+
+    // The following should be consistent with the enum in the C++ MetaspaceShared class
+    public static String[] shared_region_name = {
+        "rw",          // ReadWrite
+        "ro",          // ReadOnly
+        "bm",          // relocation bitmaps
+        "first_closed_archive",
+        "last_closed_archive",
+        "first_open_archive",
+        "last_open_archive"
+    };
+    public static int num_regions = shared_region_name.length;
+
+    public static void initialize() throws Exception {
+        WhiteBox wb = WhiteBox.getWhiteBox();
```

Comment:
This method can be changed to a `static { ....}` block, so that you don't need to explicitly call `CDSArchiveUtils.initialize()`.

Reply:
Will move to static block.